testing/integration/install_test.go

```diff
@@ -167,6 +168,68 @@ func TestInstallWithBasePath(t *testing.T) {
 	}
 }
 
+// TestRepeatedInstallUninstall will install then uninstall the agent
+// repeatedly.  This test exists because of a number of race
+// conditions that have occurred in the uninstall process.  Current
+// testing shows each iteration takes around 16 seconds.
+func TestRepeatedInstallUninstall(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		// We require sudo for this test to run
+		// `elastic-agent install` (even though it will
+		// be installed as non-root).
+		Sudo: true,
+
+		// It's not safe to run this test locally as it
+		// installs Elastic Agent.
+		Local: false,
+	})
+
+	maxRunTime := 2 * time.Minute
+	iterations := 100
+	for i := 0; i < iterations; i++ {
+		t.Run(fmt.Sprintf("%s-%d", t.Name(), i), func(t *testing.T) {
+
+			var defaultBasePath string
```

Comment:
nit: We have this logic in 3 other places so it could probably become a helper function